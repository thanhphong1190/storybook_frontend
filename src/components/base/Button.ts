import styled from 'styled-components'
import tw from 'twin.macro'

export const Button = styled.button`
  // base
  ${tw`px-4 py-4 lg:py-2 transition-colors duration-300 uppercase italic`}
  
  // primary
  ${props =>
    (!!!props.color || props.color == 'orange') &&
    tw`bg-orange hover:bg-orange__dark text-white`}
  
  // secondary
  ${props =>
    props.color == 'blue' && tw`bg-blue hover:bg-blue__dark text-white`}
  
  // white
  ${props =>
    props.color == 'white' && tw`bg-white hover:bg-grey__light text-blue`}
`

export const ButtonLink = styled.a`
  // base
  ${tw`px-4 py-4 lg:py-2 transition-colors duration-300 uppercase italic`}
  
  // primary
  ${props =>
    (!!!props.color || props.color == 'orange') &&
    tw`bg-orange hover:bg-orange__dark text-white`}
  
  // secondary
  ${props =>
    props.color == 'blue' && tw`bg-blue hover:bg-blue__dark text-white`}
  
  // white
  ${props =>
    props.color == 'white' && tw`bg-white hover:bg-grey__light text-blue`}
`

export default Button
