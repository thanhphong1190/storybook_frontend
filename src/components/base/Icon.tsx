import {
  FontAwesomeIcon,
  FontAwesomeIconProps
} from '@fortawesome/react-fontawesome'
import {
  faFacebook,
  faYoutubeSquare,
  faTwitter,
  faInstagramSquare
} from '@fortawesome/free-brands-svg-icons'
import { faStar } from '@fortawesome/free-solid-svg-icons'
export const Icons = {
  facebook: faFacebook,
  twitter: faTwitter,
  instagram: faInstagramSquare,
  youtube: faYoutubeSquare,
  star: faStar
}

export const Icon = (props: FontAwesomeIconProps) => {
  return <FontAwesomeIcon {...props} />
}
export default Icon
