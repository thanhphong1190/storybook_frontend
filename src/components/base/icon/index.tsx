export { default as SVGIcon, IconEnums } from './svgIcon'
