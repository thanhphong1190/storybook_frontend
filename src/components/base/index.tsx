export { Button, ButtonLink } from './Button'
export { SVGIcon, IconEnums } from './icon/svgIcon'
export { Icon, Icons } from './Icon'
