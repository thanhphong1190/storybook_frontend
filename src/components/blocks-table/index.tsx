export { BlocksTable, BlocksTableProps } from './BlocksTable'
