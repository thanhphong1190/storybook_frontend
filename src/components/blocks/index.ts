export { ContentBlock, ContentBlocksProps } from './ContentBlock'
export { AuthorBlock, AuthorBlockProps } from './AuthorBlock'
export { VideoBlock, PlayButton, VideoBlocksProps, VideoPlayer, VideoPlayerProps } from './VideoBlock'