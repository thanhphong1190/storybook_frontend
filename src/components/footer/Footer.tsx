import * as S from './Footer.styled'
import * as G from '../../styles/global.styles'
import { Link, Iconlink } from '../../services/api/types'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faKey } from '@fortawesome/free-solid-svg-icons'

export interface FooterProps {
  logo: string
  copyright: string
  links: Link[]
  social: {
    label: string
    socialLinks: Iconlink[]
  }
}
export const Footer = (props: FooterProps) => {
  const { logo, copyright, links, social } = props
  return (
    <S.FooterStyled>
      <S.Container>
        <S.Logo src={logo} />
        <S.InforGroup>
          <span>{copyright}</span>
          <S.LinkGroup>
            {links &&
              links.map((l, i) => (
                <>
                  <G.Link key={i} href={l.url}>
                    {l.label == 'Login' && (
                      <>
                        <FontAwesomeIcon icon={faKey} />{' '}
                      </>
                    )}
                    {l.label}
                  </G.Link>
                  {i < links.length - 1 ? ' - ' : null}
                </>
              ))}
          </S.LinkGroup>
        </S.InforGroup>
        <S.SocialGroup>
          <S.SocialLabel>{social.label}</S.SocialLabel>
          <S.SocialIcons>
            {social.socialLinks &&
              social.socialLinks.map((l, i) => (
                <G.Link key={i} href={l.url}>
                  <S.Icon name={l.icon} />
                </G.Link>
              ))}
          </S.SocialIcons>
        </S.SocialGroup>
      </S.Container>
    </S.FooterStyled>
  )
}

export default Footer
