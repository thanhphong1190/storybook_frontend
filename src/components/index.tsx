export { ContentBlock } from './blocks/ContentBlock'
export { Header } from './header/Header'
export { Navigation } from './navigation/Navigation'
export { HeroImage } from './hero/HeroImage'
export { KeyMessage } from './key-message/KeyMessage'
export { Footer } from './footer/Footer'
export { default as Banner } from './banner/Banner'
export { default as OrangeBanner } from './orange-banner/OrangeBanner'
export { ReviewCard, CustomerReviews } from './customer-reviews'
export { StandardContent } from './standard-content'
export { FranchiseBenefitCard, FranchiseBenefits } from './franchise-benefits'
export { Hero } from './hero/Hero'
export { ReadyStartedBlocks } from './ready-started-blocks'
export { BlocksTable } from './blocks-table'
