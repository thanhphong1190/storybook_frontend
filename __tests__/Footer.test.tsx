import { Footer, FooterProps } from '../src/components/footer/Footer'
import renderer from 'react-test-renderer'
import { IconEnums } from '../src/components/base'

const props: FooterProps = {
  logo: '../src/static/imgs/logo_white.png',
  copyright: 'Copyright Â© 2017 Plus Fitness. All rights reserved.',
  links: [
    { label: 'Terms and Conditions', url: '#' },
    { label: 'Advertise with us', url: '#' },
    { label: 'Privacy Policy', url: '#' },
    { label: 'Login', url: '#' }
  ],
  social: {
    label: 'Follow Us',
    socialLinks: [
      { icon: IconEnums['facebook-c'], url: '#' },
      {
        icon: IconEnums['instagram-c'],
        url: '#'
      },
      { icon: IconEnums['youtube-c'], url: '#' }
    ]
  }
}

it('renders correctly', () => {
  const tree = renderer.create(<Footer {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
