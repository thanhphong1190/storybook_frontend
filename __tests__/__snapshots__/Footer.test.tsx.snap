// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="Footerstyled__FooterStyled-vr4dm-0 fQRnVN"
>
  <div
    className="Footerstyled__Container-vr4dm-1 ftXGid"
  >
    <img
      className="Footerstyled__Logo-vr4dm-2 kbsODg"
      src="../src/static/imgs/logo_white.png"
    />
    <div
      className="Footerstyled__InforGroup-vr4dm-3 hUOXFF"
    >
      <span>
        Copyright Â© 2017 Plus Fitness. All rights reserved.
      </span>
      <div
        className="Footerstyled__LinkGroup-vr4dm-4 bqnXPN"
      >
        <a
          className="globalstyles__Link-sc-1nc69wi-2 hZljBz"
          href="#"
        >
          Terms and Conditions
        </a>
         - 
        <a
          className="globalstyles__Link-sc-1nc69wi-2 hZljBz"
          href="#"
        >
          Advertise with us
        </a>
         - 
        <a
          className="globalstyles__Link-sc-1nc69wi-2 hZljBz"
          href="#"
        >
          Privacy Policy
        </a>
         - 
        <a
          className="globalstyles__Link-sc-1nc69wi-2 hZljBz"
          href="#"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-key fa-w-16 "
            data-icon="key"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 176.001C512 273.203 433.202 352 336 352c-11.22 0-22.19-1.062-32.827-3.069l-24.012 27.014A23.999 23.999 0 0 1 261.223 384H224v40c0 13.255-10.745 24-24 24h-40v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24v-78.059c0-6.365 2.529-12.47 7.029-16.971l161.802-161.802C163.108 213.814 160 195.271 160 176 160 78.798 238.797.001 335.999 0 433.488-.001 512 78.511 512 176.001zM336 128c0 26.51 21.49 48 48 48s48-21.49 48-48-21.49-48-48-48-48 21.49-48 48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          Login
        </a>
      </div>
    </div>
    <div
      className="Footerstyled__SocialGroup-vr4dm-5 dESgeu"
    >
      <span
        className="Footerstyled__SocialLabel-vr4dm-6 kArwVu"
      >
        Follow Us
      </span>
      <div
        className="Footerstyled__SocialIcons-vr4dm-7 Acupz"
      >
        <a
          className="globalstyles__Link-sc-1nc69wi-2 hZljBz"
          href="#"
        >
          <span
            className="Iconstyles__StyledSVGIcon-sc-1cwmlu4-0 cSbqKd Footerstyled__Icon-vr4dm-8 homSvU"
          />
        </a>
        <a
          className="globalstyles__Link-sc-1nc69wi-2 hZljBz"
          href="#"
        >
          <span
            className="Iconstyles__StyledSVGIcon-sc-1cwmlu4-0 cSbqKd Footerstyled__Icon-vr4dm-8 homSvU"
          />
        </a>
        <a
          className="globalstyles__Link-sc-1nc69wi-2 hZljBz"
          href="#"
        >
          <span
            className="Iconstyles__StyledSVGIcon-sc-1cwmlu4-0 cSbqKd Footerstyled__Icon-vr4dm-8 homSvU"
          />
        </a>
      </div>
    </div>
  </div>
</section>
`;
